### PR TITLE
pythonPackages.pySMART_smartx: init at 0.3.10

### PR DESCRIPTION
**Pull request by marenz2569**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by risicle** [comment]
Two problems:

1. This package seems to implicitly depend on `smartctl` being present in the user's `PATH`. To get it properly bound to its dependencies it'll probably require doing a bit of string replacement in the source looking for places it just does a `Popen('smartctl ...`. There are a few of them.

2. Tests don't actually run. Tests require pytest and will probably need a custom `checkPhase` `cd`'ing into `pySMART/test` and running `pytest`.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -1040,6 +1040,8 @@ in {
     inherit (pkgs) cmake qt5 ninja;
   });
 
+  pySMART_smartx = callPackage ../development/python-modules/pySMART_smartx { };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: the attribute name needs to be a normalized name, so `pysmart-smartx`.

File: pkgs/development/python-modules/pySMART_smartx/default.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{ stdenv, buildPythonPackage, fetchPypi
+}:
+
+buildPythonPackage rec {
+  pname = "pySMART.smartx";
```
Comment (FRidh) [inline pkgs/development/python-modules/pySMART_smartx/default.nix]: use normalized name here as well

File: pkgs/development/python-modules/pySMART_smartx/default.nix [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  version = "0.3.10";
+
+  src = fetchPypi {
+    inherit pname version;
```
Comment (FRidh) [inline pkgs/development/python-modules/pySMART_smartx/default.nix]: and here `pname = "pySMART.smartx"`;

**Review by FRidh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**